### PR TITLE
Adding `tbl_strata2`

### LINKED ISSUE #1091: Add `tbl_strata()` and `tbl_strata2()` set like `map()` and `map2()`
The second `tbl_strata2()` function will pass the subset data frame AND the strata name.

We could add information like that found in `df_by()`, use glue to construct that stratum label that could include information like sample size etc.

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**

* New function `tbl_strata2()` that passes both the the stratified data frame as well as the stratum level to the user function. (#/1091)
* Added `tbl_strata(.header=)` argument providing greater control over the stratum headers that are added to the tables.

**If there is an GitHub issue associated with this pull request, please provide link.**
closes #1091

QUESTION: We added an argument to control the header and to be able to add Ns to the header as well. Should this header be passed as the second argument to the `.tbl_fun=` function, or should the stratifying level be passed along? For `tbl_strata(strata = trt, .header = "**{strata}**, N = {n}")`, this would be the difference between passing `"Drug A"` and `"Drug A, N = 98"`. What do you think?

--------------------------------------------------------------------------------

Checklist for PR reviewer

- [x] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [ ] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [ ] If a new function was added, function included in `_pkgdown.yml`
- [ ] If a bug was fixed, a unit test was added for the bug check
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features. Review coverage with `covr::report()`. Before you run, set `Sys.setenv(NOT_CRAN="true")` and begin in a fresh R session without any packages loaded. 
- [ ] R CMD Check runs without errors, warnings, and notes
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update NEWS.md with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parantheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -6,8 +6,9 @@
 
 --------------------------------------------------------------------------------
 
-Checklist for PR reviewer
+Reviewer Checklist (if item does not apply, mark is as complete)
 
+- [ ] Ensure all package dependencies are installed by running `renv::install()`
```
Comment (jalavery) [inline .github/pull_request_template.md:11]: thank you for adding this!

**Review by jalavery — COMMENTED** [review]
Is it possible to add a second example showing when you would use `tbl_strata2()` instead of `tbl_strata()`?

**Comment by ddsjoberg** [comment]
> Is it possible to add a second example showing when you would use `tbl_strata2()` instead of `tbl_strata()`?

@jalavery good point! example added!

**Comment by jalavery** [comment]
> @jalavery QUESTION: We added an argument to control the header and to be able to add Ns to the header as well. Should this header be passed as the second argument to the .tbl_fun= function, or should the stratifying level be passed along? For tbl_strata(strata = trt, .header = "**{strata}**, N = {n}"), this would be the difference between passing "Drug A" and "Drug A, N = 98". What do you think?

I think I like "Drug A, N = 98"

**Comment by ddsjoberg** [comment]
> > @jalavery QUESTION: We added an argument to control the header and to be able to add Ns to the header as well. Should this header be passed as the second argument to the .tbl_fun= function, or should the stratifying level be passed along? For tbl_strata(strata = trt, .header = "**{strata}**, N = {n}"), this would be the difference between passing "Drug A" and "Drug A, N = 98". What do you think?
> 
> I think I like "Drug A, N = 98"

@jalavery thanks! then no updated needed, that is what is being passed in the second arg of `tbl_strata2()`

**Review by jalavery — APPROVED** [review]

**Pull request merged**